durn/actions.go

```diff
@@ -11,24 +10,88 @@ import (
 	"durn2.0/util"
 )
 
-// AddValidVoters checks that provided voters are entered as valid emails, and if that
-// is the case inserts them into the database
-func AddValidVoters(ctx context.Context, voters []models.Voter) error {
+// AddValidVoters checks that provided voters are entered as valid emails, and
+// if that is the case inserts them into the database.
+// Returns all voters for which it failed
+func AddValidVoters(ctx context.Context, voters []models.Voter) ([]models.Voter, error) {
+	dbConn := db.TakeDB()
+	defer db.ReleaseDB()
+
 	mailRegex := "[^@]+@kth\\.se"
+	votersToInsert := []db.ValidVoter{}
```

Comment:
I'm not sure I quite understand the reason why but this is not the preferred way to [declare an empty slice](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices), apparently:
 
```go
var votersToInsert []db.ValidVoter
```
is the preferred way but does return `"null"` instead of `"[]"` when converted to JSON. I can see you are doing that in other places.

Reply:
resonable, will fix

Reply:
Yeah, I had a closer look at what is going on and it does seem like a slice is basically 3 values, `ptr`, `len`, `cap`, (pointer, length, and capacity). In the case of a `nil` slice the value of `ptr` is `nil` and in the other case it is pointing to an empty array somewhere in memory.

Reply:
Exactly, the reason for preferring `var name []type` seems to be in order to avoid unnecessary allocation, from what I can see.